osquery/tables/system/posix/xattr_utils.cpp

```diff
@@ -0,0 +1,189 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifndef __FreeBSD__
+
+#include <string>
+#include <vector>
+
+#include <sys/xattr.h>
+
+#include <osquery/logger.h>
+
+#include "osquery/tables/system/posix/xattr_utils.h"
+
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+
+#ifndef TARGET_OS_MAC
+#error Unsupported macOS target
+#endif
+
+#include "osquery/tables/system/darwin/special_xattr_decoder.h"
+
+#else
+#include "osquery/tables/system/linux/special_xattr_decoder.h"
+#endif
+
+namespace osquery {
+namespace {
+// Compatibility wrappers for macOS
+#if defined(TARGET_OS_MAC)
+ssize_t getxattr(const char* path, const char* name, void* value, size_t size) {
+  return ::getxattr(path, name, value, size, 0, 0);
+}
+
+ssize_t listxattr(const char* path, char* list, size_t size) {
+  return ::listxattr(path, list, size, 0);
+}
+
+int setxattr(const char* path,
```

Comment:
I don't want to support setting xattrs

Reply:
This is only used for the tests; the wrappers are there so that we can have a single `extended_attributes_tests.cpp` for all platform (FreeBSD/Linux/macOS)

Reply:
Here's the original code that sets the extended attributes for the tests: https://github.com/facebook/osquery/blob/master/osquery/tables/system/darwin/tests/extended_attributes_tests.cpp#L70

I have now added a comment above the function:
``` cpp
// Used by the tests in osquery/tables/system/posix/tests/extended_attributes_tests.cpp
bool setExtendedAttributes(
    const std::string& path,
    const std::unordered_map<std::string, std::string>& attributes) {
```